lib/solidus_paypal_braintree/engine.rb

```diff
@@ -19,9 +19,11 @@ class Engine < Rails::Engine
     end
 
     initializer "register_solidus_paypal_braintree_gateway", after: "spree.register.payment_methods" do |app|
-      app.config.spree.payment_methods << SolidusPaypalBraintree::Gateway
-      SolidusPaypalBraintree::Gateway.allowed_admin_form_preference_types << :preference_select
-      Spree::PermittedAttributes.source_attributes.concat [:nonce, :payment_type, :paypal_funding_source]
+      config.to_prepare do
+        app.config.spree.payment_methods << SolidusPaypalBraintree::Gateway
```

Comment:
This `append` call is already idempotent since `app.config.spree.payment_methods` is a Set object